warmup.py

```diff
@@ -240,17 +251,39 @@ def accumulate(self, L, points):
         The sample is accumulated into the radial distribution average.
 
         """
-        pass
+        if self.rmax >= L/2:
+            raise ValueError('rmax >= L/2')
+        temp_rdf = np.zeros_like(self._rdf)
+        for i, point in enumerate(points[:-1]):
+            neighbors = np.array(points[i+1:])
+            for neighbor in neighbors:
+                dist_vect = np.absolute(point - neighbor)
+                dist_vect -= L*np.round(dist_vect/L) 
+                dist_mag = np.linalg.norm(dist_vect)
+                if self.rmin <= dist_mag <= self.rmax:
+                    for j, bin_dist in enumerate(self.centers):
+                        r_outer = self.edges[j+1]
+                        r_inner = self.edges[j]
+                        if r_inner < dist_mag <= (r_outer + 1e-8):
+                            temp_rdf[j] += 2
+        rho = len(points)/L**3
```

Comment:
Once you've coerced points to a numpy array, you can use points.shape[0], which is probably a little clearer.